x-pack/legacy/plugins/siem/public/pages/detection_engine/detection_engine.tsx

```diff
@@ -131,7 +135,7 @@ const DetectionEnginePageComponent: React.FC<DetectionEnginePageComponentProps>
   return (
     <>
       {hasIndexWrite != null && !hasIndexWrite && <NoWriteSignalsCallOut />}
-      <WithSource sourceId="default">
+      <WithSource sourceId="default" indexToAdd={indexToAdd}>
```

Comment:
Nice fix! Verified autocomplete now works for `signal` on the main DE page, Rule Details, and also within timeline when `All events` or `Signal events` is selected. Looks good 👍